examples/run_Hahn_echo.py

```diff
@@ -1,129 +1,95 @@
-'''Hahn Echo Experiment
-=======================
- To run this experiment, please open Xepr on the EPR computer, connect to
- spectrometer, load the experiemnt 'set_field' and enable XEPR API. Then, in a
- separate terminal, run the program XEPR_API_server.py, and wait for it to
- tell you 'I am listening' - then, you should be able to run this program from
- the NMR computer to set the field etc. 
+"""
+Spin Echo
+=========
 
- Note the booleans user_sets_Freq and
- user_sets_Field allow you to run experiments as previously run in this lab.
- If both values are set to True, this is the way we used to run them. If both
- values are set to False, you specify what field you want, and the computer
- will do the rest.
-'''
+To run this experiment, please open Xepr on the EPR computer, connect to
+spectrometer, load the experiment 'set_field' and enable XEPR API. Then, in a
+separate terminal, run the program XEPR_API_server.py, and wait for it to
+tell you 'I am listening' - then, you should be able to run this program from
+the NMR computer to set the field etc. 
+"""
 
 from pylab import *
 from pyspecdata import *
 import os
-import sys
 import SpinCore_pp
 from SpinCore_pp.ppg import run_spin_echo
 from datetime import datetime
-import numpy as np
-from Instruments import power_control
 from Instruments.XEPR_eth import xepr
-import h5py
 fl = figlist_var()
-p90_us = 4.477
+#{{{importing acquisition parameters
+config_dict = SpinCore_pp.configuration('active.ini')
+nPoints = int(config_dict['acq_time_ms']*config_dict['SW_kHz']+0.5)
+#}}}
+#{{{create filename and save to config file
 date = datetime.now().strftime('%y%m%d')
-#{{{these parameters change for each sample
-output_name = 'test'
-node_name = 'echo'
-adcOffset = 28
-user_sets_Freq = True
-user_sets_Field = True
-#{{{ set field here
-if user_sets_Field:
-    # You must enter field set on XEPR here
-    true_B0 = 3424.42
-    print("My field in G should be %f"%true_B0)
+config_dict['type'] = 'echo'
+config_dict['date'] = date
+config_dict['echo_counter'] += 1
+filename = str(config_dict['date'])+'_'+config_dict['chemical']+'_'+config_dict['type'])
 #}}}
 #{{{let computer set field
-if not user_sets_Field:
-    desired_B0 = 3506.50
-    with xepr() as x:
-        true_B0 = x.set_field(desired_B0)
-    print("My field in G is %f"%true_B0)
-#}}}
-#{{{ set frequency here
-if user_sets_Freq:
-    carrierFreq_MHz = 14.8978438
-    print("My frequency in MHz is",carrierFreq_MHz)
-#}}}
-#{{{ let computer set frequency
-if not user_sets_Freq:
-    gamma_eff = (14.897706/3506.5)
-    carrierFreq_MHz = gamma_eff*true_B0
-    print("My frequency in MHz is",carrierFreq_MHz)
+print("I'm assuming that you've tuned your probe to",
+        config_dict['carrierFreq_MHz'],
+        "since that's what's in your .ini file")
+print("Based on your .ini file, I'm setting the field to %f"%config_dict['Field'])
```

Comment:
changing the gamma_eff takes more calculations for myself when trying to get on resonance that simply changing the field. THEN once on resonance I can input the actual gamma effective